core/src/consensus/light_client/mod.rs

```diff
@@ -0,0 +1,87 @@
+// Copyright 2020 Kodebox, Inc.
+// This file is part of CodeChain.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pub mod verification;
+pub use self::verification::verify_header;
+use ckey::SchnorrSignature;
+pub use ctypes::BlockNumber;
+use ctypes::{CompactValidatorSet, Header};
+pub use primitives::{H256, H512};
+use rlp::{Decodable, DecoderError, Encodable, Rlp, RlpStream};
+
+#[derive(PartialEq, Debug)]
+// (index_in_vset, sign); schnorr scheme
+pub struct Seal(Vec<(usize, SchnorrSignature)>);
+
+impl Encodable for Seal {
+    fn rlp_append(&self, s: &mut RlpStream) {
+        s.begin_list(self.0.len() * 2);
+        for (x, y) in self.0.iter() {
+            s.append(x);
+            s.append(&H512::from(*y));
+        }
```

Comment:
`SchnorrSignature` is implementing `Encodable`. Why did you convert the signature to `H512`?

Reply:
I don't know why github doesn't update this but it's `Outdate`